src/types/CodepointWidthDetector.cpp

```diff
@@ -325,68 +331,793 @@ namespace
+    0x40, 0x40, 0x40, 0x40, 0x02, 0x02, 0x02, 0x40,
+    0x02, 0x02, 0x40, 0x40, 0x89, 0x8a, 0x8a, 0x8a,
+    0x8a, 0x8a, 0x8a, 0x8a, 0x8a, 0x89, 0x8a, 0x8a,
+    0x8a, 0x8a, 0x40, 0x40, 0x40, 0x40, 0x47, 0x47,
+    0x47, 0x47, 0x47, 0x47, 0x47, 0x40, 0x40, 0x40,
+    0x48, 0x48, 0x48, 0x48, 0x48, 0x40, 0x40, 0x40,
+    0x40, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02,
+    0x82, 0x40, 0x40, 0x40, 0x40, 0x40, 0x40, 0x40,
+    0x01, 0x40, 0x40, 0x40, 0x40, 0x40, 0x40, 0x42,
+    0x42, 0x41, 0x01, 0x01, 0x01, 0x40, 0xc0, 0x40,
+    0x40, 0x02, 0x02, 0x02, 0x40, 0x40, 0x03, 0x40,
+    0x40, 0x02, 0x40, 0x43, 0x43, 0x40, 0x40, 0x40,
+    0x40, 0x02, 0x02, 0x02, 0x02, 0x40, 0x02, 0x02,
+    0x40, 0x40, 0x02, 0x02, 0x40, 0x40, 0x02, 0x02,
+    0x02, 0x02, 0x43, 0x02, 0x02, 0x40, 0x40, 0x40,
+    0x40, 0x02, 0x02, 0x43, 0x02, 0x02, 0x02, 0x02,
+    0x40, 0x40, 0x40, 0x40, 0x43, 0x43, 0x43, 0x43,
+    0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x40, 0x02,
+    0x02, 0x02, 0x02, 0x02, 0x02, 0x40, 0x40, 0x02,
+    0x40, 0x02, 0x02, 0x40, 0x02, 0x02, 0x02, 0x02,
+    0x02, 0x02, 0x43, 0x02, 0x40, 0x40, 0x40, 0x40,
+    0x80, 0x80, 0x80, 0x80, 0x02, 0x40, 0x40, 0x40,
+    0x80, 0x80, 0x80, 0x80, 0x40, 0x80, 0x80, 0x40,
+    0x40, 0x80, 0x40, 0x40, 0x40, 0x40, 0x40, 0x80,
+    0x80, 0x80, 0x40, 0x40, 0x80, 0x40, 0x40, 0x01,
+    0x01, 0x01, 0x01, 0x40, 0x40, 0x40, 0x40, 0x02,
+    0x02, 0x02, 0x01, 0x01, 0x01, 0x01, 0x01, 0x02,
+    0x02, 0x02, 0x02, 0x02, 0x40, 0x40, 0x02, 0x02,
+    0x02, 0x02, 0x02, 0x4c, 0x4c, 0x4c, 0x4c, 0x8c,
+    0x4c, 0x4c, 0x4c, 0xc0, 0xc0, 0xc0, 0x40, 0x40,
+    0x4c, 0x4c, 0x4c, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0,
+    0xc0, 0x40, 0x4c, 0xc0, 0xc0, 0x40, 0x40, 0x4c,
+    0x4c, 0x4c, 0x4c, 0xcc, 0xcc, 0xc0, 0xc0, 0xc0,
+    0xc0, 0xc0, 0xc0, 0xcc, 0xcc, 0xc0, 0xc0, 0xc0,
+    0xc0, 0xc0, 0xc0, 0x8c, 0xc0, 0x8c, 0x8c, 0x8c,
+    0x8c, 0x8c, 0x8c, 0x8c, 0xc0, 0xc0, 0xc0, 0xc0,
+    0xc0, 0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x45,
+    0x45, 0x45, 0x45, 0x45, 0x45, 0x45, 0x45, 0x80,
+    0x8c, 0x8c, 0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x80,
+    0x80, 0x8c, 0x80, 0x80, 0x80, 0x80, 0x80, 0x8c,
+    0x8c, 0x8c, 0x8c, 0x8c, 0x8c, 0x80, 0x4c, 0x4c,
+    0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x8c, 0x8c, 0x4c,
+    0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x8c, 0x8c,
+    0x8c, 0x8c, 0x8c, 0x8c, 0x4c, 0x8c, 0x8c, 0x8c,
+    0x4c, 0x4c, 0x4c, 0x4c, 0x8c, 0x4c, 0x4c, 0x4c,
+    0x8c, 0x4c, 0x4c, 0x4c, 0x8c, 0x8c, 0x8c, 0x82,
+    0x82, 0x82, 0x82, 0x82, 0x8c, 0x4c, 0x8c, 0x8c,
+    0x8c, 0x8c, 0x8c, 0x8c, 0x4c, 0x4c, 0x8c, 0x8c,
+    0x8c, 0x8c, 0x8c, 0x8c, 0x40, 0x40, 0x4c, 0x4c,
+    0x4c, 0x8c, 0x8c, 0x8c, 0x8c, 0x4c, 0x40, 0x40,
+    0x40, 0x40, 0x4c, 0x4c, 0x4c, 0x4c, 0x4c, 0x4c,
+    0x40, 0x40, 0x40, 0x40, 0x8c, 0x8c, 0x8c, 0x8c,
+    0x40, 0x8c, 0x8c, 0x8c, 0x8c, 0x8c, 0x8c, 0x40,
+    0x8c, 0x41, 0x01, 0x41, 0x41, 0x41, 0x41, 0x41,
+    0x41, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0, 0x40,
+    0x40,
+};
+static constexpr uint8_t s_joinRules[13][13] = {

```

Comment:
is there value in packing this into 13 uint16_t?

Reply:
You're a goddamn genius.